src/RestEase/RequestPathParamSerializer.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+
+namespace RestEase
+{
+    /// <summary>
+    /// Helper which knows how to serialize path parameters
+    /// </summary>
+    public abstract class RequestPathParamSerializer
+    {
+        /// <summary>
+        /// Serialize a path parameter whose value is scalar (not a collection), into a name -> value pair
+        /// </summary>
+        /// <param name="name">Name of the path parameter</param>
+        /// <param name="value">Value of the path parameter</param>
+        /// <param name="info">Extra info which may be useful to the serializer</param>
+        /// <returns>A name -> value pair to use as path parameter</returns>
+        public abstract KeyValuePair<string, string> SerializePathParam(string name,object value, RequestPathParamSerializerInfo info);
```

Comment:
Argh sorry, I didn't notice this before. I think this should only return a `string`.

For query parameters, returning multiple key/value pairs makes sense, However for path parameters, changing the name of the placeholder being substituted doesn't make sense - if the implementation of this method does not return `name` as the `key` in the `KeyValuePair`, then breakage will happen, and that's a bad API.

The `PathParameterInfo` should take the string return value of the serializer, and add in `this.name`.

Reply:
Ah yes that makes a lot more sense. I think the use of `KeyValuePair<string, string>` elsewhere in `PathParameterInfo` threw me off, but its use is different there.

Reply:
Fixed this in latest commits.

Reply:
Oh! And the method should be generic, to match the other serializers, with `value` being a `T`.

Reply:
Pushed new commits now to make it generic.